components/_impl/tasks/base.py

```diff
@@ -119,16 +120,28 @@ def foo() -> None:
     src_lines = f_src.splitlines(keepends=False)
 
     node: ast.AST
-    for node in f_ast.body[0].body:
+    for idx, node in enumerate(f_ast.body[0].body):
         # In Python 3.7, there is a bug in `ast` that causes it to incorrectly
         # report the start line of bare multi-line strings:
         #   https://bugs.python.org/issue16806
         # Given that the only use for such strings is a docstring (or multi
         # line comment), we simply elect to skip over them and index on the
         # first node that will give valid indices.
-        if node.col_offset == -1:
-            assert isinstance(node.value, ast.Str), f"Expected `ast.Str`, got {type(node)}. ({node}) {node.lineno}"
-            continue
+        # To support both Python 3.7 and 3.8,
```

Comment:
I don't think we need to change this. Stripping docstrings won't affect the execution in 3.8+.

Reply:
Just to confirm, do you mean we don't strip docstrings for both 3.7 and 3.8? Sounds good to me, and I will go ahead and change the test result in https://github.com/pytorch/benchmark/blob/main/components/test/test_subprocess.py#L88.

Reply:
Oh, I see what you mean. I baked the 3.7 behavior into the test. Yeah, you can just remove the docstring check from the test.

Reply:
I kinda feel bad if we don't cover docstring in our test case, so I use different test results for different python versions. Please help review, thanks!

Reply:
We can also use an f-string predicated on Python version. The only down side is that we don't get the auto-update from expecttest.

Reply:
I agree, it is kind of sad to lose test coverage.

Reply:
Thanks for the suggestion, improved test using f-string.